typescript/src/lib/util/functions/channels.ts

```diff
@@ -0,0 +1,24 @@
+import { Permissions, GuildChannel, Message } from 'discord.js';
+
+export type ChannelTypes = Message['channel'];
+export type GuildBasedChannelTypes = Extract<ChannelTypes, GuildChannel>;
```

Comment:
Are these types really necessary? I can smell them being duplicated in various places coming.